app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/MapViewModel.kt

```diff
@@ -29,18 +39,40 @@ constructor(val locationSource: LocationSource, private val beaconRepository: Be
   private val _uiState = MutableStateFlow(BeaconListUiState(loading = true))
   val uiState: StateFlow<BeaconListUiState> = _uiState
 
+  private var _beaconList = MutableStateFlow<List<Beacon>>(emptyList())
+  val beaconList: StateFlow<List<Beacon>> = _beaconList
+
+  private val _areBeaconsLoaded = MutableStateFlow(false)
+
   init {
     observeBeacons()
   }
 
   private fun observeBeacons() {
     viewModelScope.launch {
       beaconRepository.getAll().collect { beacons ->
+        _beaconList.value = beacons // Ensure `beaconList` is updated
         _uiState.value = BeaconListUiState(beacons = beacons, loading = false)
       }
     }
   }
 
+  private fun retrieveBeacons(location: Location1, context: Context) {
+    viewModelScope.launch {
+      // Update _uiState to reflect a loading state
+      _uiState.value = BeaconListUiState(loading = true)
+
+      createNearbyBeacons(
+          location, _beaconList, 10000.0, context, beaconRepository, viewModelScope) {
+            val updatedBeacons =
+                _beaconList
+                    .value // +  placeBeaconsRandomly(_beaconList.value, location, beaconRepository)
```

Comment:
Is this comment still relevant ? If yes I think it would be good to add that line, else just remove it please.

Reply:
not relevant for now, i will remove it 🫡

Reply:
I think you forgor to remove it